### PR TITLE
Added Project Euler Problem #3 in JS

### PR DESCRIPTION
**Pull request by arnavb**
**Fixes issue:** N/A

**Changes:**
- Added Project Euler problem #3 in JS.

### CODE REVIEW

File: code/online_challenges/src/project_euler/problem_003/problem_003.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+let num = 600851475143;
+let highestPrimeFactor = 0;
+let currentFactor = 2;
+
+while (num !== 1) {
```
Comment (abdouskamel) [inline code/online_challenges/src/project_euler/problem_003/problem_003.js]: There are cases where num would never be equal to 1, for example num = 4.

File: code/online_challenges/src/project_euler/problem_003/problem_003.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  if (num % currentFactor === 0 && currentFactor > highestPrimeFactor) {
+    highestPrimeFactor = currentFactor;
```
Comment (abdouskamel) [inline code/online_challenges/src/project_euler/problem_003/problem_003.js]: currentFactor could be not a prime, for example for num = 8, in the last iteration we get currentFactor = 4.

Comment (arnavb) [reply code/online_challenges/src/project_euler/problem_003/problem_003.js]: You are right, I should reset the factor at some point.

**Comment by arnavb** [comment]
@abdouskamel Could you review if the algorithm is correct now?

**Comment by abdouskamel** [comment]
@arnavb It still gives 4 for entry 8. The problem is that there's the possibility to affect to `highestPrimeFactor` every divisor of `num` that we found. In fact, we should affect to the first variable a number only if it's a prime.

**Comment by arnavb** [comment]
@abdouskamel You were entirely correct. However, I found another more efficient algorithm that achieves the same purpose. Please review now! Thanks for your effort!

**Comment by arnavb** [comment]
@abdouskamel The checking for that could reduce the performance of the algorithm, so I will leave this as it is. The user should make sure that they pass the correct number.

@AdiChat please merge!

**Comment by arnavb** [comment]
@AdiChat please look at my LGTM PR!

**Comment by AdiChat** [comment]
Sure, I will take a look at it now :+1: 

Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**